sdk/idp_oauth_access_token_source.go

```diff
@@ -0,0 +1,60 @@
+package sdk
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/opentdf/platform/lib/ocrypto"
+	"github.com/opentdf/platform/sdk/auth"
+	"golang.org/x/oauth2"
+)
+
+// OAuthAccessTokenSource allow connecting to an IDP and obtain a DPoP bound access token
+type OAuthAccessTokenSource struct {
+	source         oauth2.TokenSource
+	scopes         []string
+	dpopKey        jwk.Key
+	asymDecryption ocrypto.AsymDecryption
+	dpopPEM        string
+}
+
+func NewOAuthAccessTokenSource(
+	source oauth2.TokenSource, scopes []string, key *ocrypto.RsaKeyPair,
+) (*OAuthAccessTokenSource, error) {
+	dpopPublicKeyPEM, dpopKey, asymDecryption, err := getNewDPoPKey(key)
+	if err != nil {
+		return nil, err
+	}
+
+	tokenSource := OAuthAccessTokenSource{
+		source:         source,
+		scopes:         scopes,
+		asymDecryption: *asymDecryption,
+		dpopKey:        dpopKey,
+		dpopPEM:        dpopPublicKeyPEM,
+	}
+
+	return &tokenSource, nil
+}
+
+// AccessToken use a pointer receiver so that the token state is shared
+func (t *OAuthAccessTokenSource) AccessToken(_ context.Context, _ *http.Client) (auth.AccessToken, error) { // must satisfy auth.AccessTokenSource interface
+	tok, err := t.source.Token()
+	if err != nil {
+		return "", fmt.Errorf("error getting access token: %w", err)
+	}
+
+	if tok.Expiry.Before(time.Now()) {
```

Comment:
```suggestion
	if !tok.Valid() {
```

It looks like the valid check does some additional checks (token has been set, and the expiration has been set as well).  The expiration check also allows a small time buffer.